### DISCUSSION TITLE
How to connect Standart .NET libs ?

### DISCUSSION THREAD

**Discussion by ADRNV** [Brick-Abode/pldotnet#11] [Q&A]
Hi. I want to make HttpRequest to some resource for this need `System.Net.Http;` But i cant use `using`  C# statement in PL.NET otherwise or without this statement not works. How i can make this ?Or in this moment it is imposible and need to extend PL.NET ? Can i extend this by analogy with `Elog` ?
Same required for WCF, gRPC

**Comment by houstonhaynes** [comment]
I was just thinking about this too - such as bringing in GeoSpacial libraries, etc. Inquiring minds want to know! :)

**Comment by tlewis-ba** [comment]
Ooh, this is important.  We had discussed how to do it, but I don't think we implemented it.

Let us take a look and get back to you, or feel free to submit an MR.  (If you do code it, then I suggest discussing the design here before coding, though of course you don't have to.)

**Reply by ADRNV** [reply]
Amazing. I'll try attract colleagues who knows C. How i understand C# is just wrapper(if dont attention on handlers, engine end etc. important thigs) for which it is write C functions ?

**Comment by tlewis-ba** [comment]
You won't need to touch C; this would all be C#.

You'll need to declare the needed libs somehow in your stored procedure code, and then modify the UserFunction file to do the import.  That is 100% C# code.

Want to start with a design document?  We could offer feedback.

**Reply by ADRNV** [reply]
Yes. It be useful. I want try add some features. I be thanksful for introduction to development this project

**Reply by tlewis-ba** [reply]
Free software can be the source of great joy in life, and we are honored to have you!

**Reply by ADRNV** [reply]
How I can learn design of this project ?Can you introduct to this steps of develop:

1. Environment setup
2. Build Process
3. Test process

**Reply by tlewis-ba** [reply]
@ADRNV, take a look at `INSTALL.md`, follow the instructions there, and post your questions/problems here.
